File: src/main/java/minic/ui/MiniCInspectorModel.java
package minic.ui;

import java.util.Objects;

/**
 * Inspector 面板展示数据。
 *
 * @param currentState 当前状态文本
 * @param currentItem 当前项文本
 * @param accumulatedOutput 累计输出文本
 */
public record MiniCInspectorModel(
        String currentState,
        String currentItem,
        String accumulatedOutput
) {
    public MiniCInspectorModel {
        Objects.requireNonNull(currentState, "currentState");
        Objects.requireNonNull(currentItem, "currentItem");
        Objects.requireNonNull(accumulatedOutput, "accumulatedOutput");
    }
}

File: uiweb/src/panel/MiniCInspectorModel.ts
import type { JavaMirrorFile } from "../translation/javaMirror";

export const miniCInspectorModelMirror = {
  "javaPath": "src/main/java/minic/uilocal/panel/MiniCInspectorModel.java",
  "webPath": "uiweb/src/panel/MiniCInspectorModel.ts",
  "packageName": "minic.uilocal",
  "exportName": "MiniCInspectorModel",
  "kind": "record",
  "imports": [
    "java.util.Objects"
  ],
  "fields": [],
  "methods": [
    {
      "name": "MiniCInspectorModel",
      "signature": "MiniCInspectorModel(String currentState, String currentItem, String accumulatedOutput)"
    }
  ]
} as const satisfies JavaMirrorFile;

export class MiniCInspectorModel {
  static readonly mirror = miniCInspectorModelMirror;

  readonly mirror = miniCInspectorModelMirror;

  readonly currentState: string;

  readonly currentItem: string;

  readonly accumulatedOutput: string;

  constructor(currentState: string, currentItem: string, accumulatedOutput: string) {
    this.currentState = currentState;
    this.currentItem = currentItem;
    this.accumulatedOutput = accumulatedOutput;
  }

  summary(): string {
    return `MiniCInspectorModel: ${this.mirror.methods.length} methods, ${this.mirror.fields.length} fields`;
  }
}

export default MiniCInspectorModel;
